### PR TITLE
[9.x] Deprecation Test Improvements

### PR DESCRIPTION
**Pull request by crynobone**
* Bump `fakerphp/faker` from `^1.9.2` to `^1.21`
* Bump `league/commonmark` from `^2.2` to `^2.2.1`

### CODE REVIEW

**Comment by driesvints** [comment]
@crynobone I thought we already were treating deprecations as exceptions: https://github.com/laravel/framework/blob/9.x/phpunit.xml.dist#L6

This can definitely just go into the current test suite. No need for a separate workflow 👍

**Comment by nunomaduro** [comment]
>  I thought we already were treating deprecations as exceptions

@driesvints Not in application tests that actually need to boot the framework. As Laravel boots, and HandleExceptions actually proxy those deprecations to a specific log channel, if any.

@crynobone Out of curiosity, can you point me to the exact code, on testbench, that modifies the behavior on `HandleExceptions` so deprecations actually get reported?

**Comment by crynobone** [comment]
Testbench would bootstrap `Orchestra\Testbench\Bootstrap\HandleExceptions`  insteads of Laravel.

**Comment by driesvints** [comment]
Hey @crynobone. I'm gonna take this one over and see if I can solve all issues. I think the right way to tackle prefer-lowest is to bump minimum versions. For anything we cannot solve through Composer we can use the already provided step here: 

https://github.com/laravel/framework/blob/9.x/.github/workflows/tests.yml#L84

**Comment by driesvints** [comment]
@crynobone I feel like we're still missing things here. For example, the first failing test I look at is:

```
1) Illuminate\Tests\Integration\Auth\ForgotPasswordTest::it_can_send_forgot_password_email
str_starts_with(): Passing null to parameter #1 ($haystack) of type string is deprecated

/home/runner/work/framework/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:71
/home/runner/work/framework/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:268
/home/runner/work/framework/framework/src/Illuminate/Auth/Passwords/PasswordBrokerManager.php:86
/home/runner/work/framework/framework/src/Illuminate/Auth/Passwords/PasswordBrokerManager.php:71
/home/runner/work/framework/framework/src/Illuminate/Auth/Passwords/PasswordBrokerManager.php:48
/home/runner/work/framework/framework/src/Illuminate/Support/Facades/Facade.php:338
/home/runner/work/framework/framework/tests/Integration/Auth/ForgotPasswordTest.php:53
```

But there's no `str_starts_with` call on that line. Is the real stack trace hidden somewhere?

**Comment by driesvints** [comment]
Seems like it's the case in most tests: https://github.com/laravel/framework/actions/runs/3707093739/jobs/6283094947#step:9:159

```
1) Illuminate\Tests\Integration\Auth\ForgotPasswordWithoutDefaultRoutesTest::it_can_send_forgot_password_email_via_create_url_using
Use of "static" in callables is deprecated

/home/runner/work/framework/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:71
/home/runner/work/framework/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:266
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/Provider/Base.php:357
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/Provider/Base.php:408
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/Provider/Base.php:423
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/Provider/Internet.php:94
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/Provider/Internet.php:48
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/Generator.php:230
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/Generator.php:287
/home/runner/work/framework/framework/vendor/fakerphp/faker/src/Faker/UniqueGenerator.php:48
/home/runner/work/framework/framework/vendor/orchestra/testbench-core/src/Factories/UserFactory.php:31
/home/runner/work/framework/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:446
/home/runner/work/framework/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:425
/home/runner/work/framework/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:409
/home/runner/work/framework/framework/src/Illuminate/Database/Eloquent/Concerns/GuardsAttributes.php:155
/home/runner/work/framework/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:408
/home/runner/work/framework/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:382
/home/runner/work/framework/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:276
/home/runner/work/framework/framework/tests/Integration/Auth/ForgotPasswordWithoutDefaultRoutesTest.php:78
```

File: src/Illuminate/Http/Resources/Json/AnonymousResourceCollection.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace Illuminate\Http\Resources\Json;
 
+#[\AllowDynamicProperties]
```
Comment (crynobone) [inline src/Illuminate/Http/Resources/Json/AnonymousResourceCollection.php:5]: Fixes `Creation of dynamic property Illuminate\Http\Resources\Json\AnonymousResourceCollection::$preserveKeys is deprecated`

File: src/Illuminate/Console/Scheduling/ScheduleListCommand.php [hunk 2]
```diff
@@ -109,7 +112,9 @@ public function handle(Schedule $schedule)
                 ? $nextDueDate->format('Y-m-d H:i:s P')
                 : $nextDueDate->diffForHumans();
 
-            $hasMutex = $event->mutex->exists($event) ? 'Has Mutex › ' : '';
+            $hasMutex = $event instanceof CallbackEvent
+                ? (is_string($event->description) && $event->mutex->exists($event) ? 'Has Mutex › ' : '')
+                : ($event->mutex->exists($event) ? 'Has Mutex › ' : '');
```
Comment (crynobone) [inline src/Illuminate/Console/Scheduling/ScheduleListCommand.php:117]: `CallbackEvent` depends on unique `$event->description` to generate mutex cache key and it is optional.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -73,7 +73,7 @@ jobs:
         with:
           timeout_minutes: 5
           max_attempts: 5
-          command: composer require ramsey/collection:^1.2 brick/math:^0.9.3 --no-interaction --no-update
+          command: composer require ramsey/collection:^1.2 brick/math:^0.9.3 symfony/css-selector:^6.0 --no-interaction --no-update
```
Comment (crynobone) [inline .github/workflows/tests.yml:76]: without this, `tijsverkoyen/css-to-inline-styles` allows requiring `symfony/css-selector` v2 with `--prefer-lowest`

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -145,12 +145,20 @@ jobs:
...
+        if: matrix.php >= 8.1
+
       - name: Set Minimum PHP 8.2 Versions
         uses: nick-fields/retry@v2
         with:
           timeout_minutes: 5
           max_attempts: 5
-          command: composer require guzzlehttp/guzzle:~7.5 guzzlehttp/psr7:~2.4 predis/predis:^2.0.2 --no-interaction --no-update
+          command: composer require guzzlehttp/guzzle:~7.5 guzzlehttp/psr7:~2.4 predis/predis:~2.0.2 --no-interaction --no-update
         if: matrix.php >= 8.2
```
Comment (crynobone) [inline .github/workflows/tests.yml:162]: We can add this or completely ignores deprecation errors on Windows build by adding `TESTBENCH_CONVERT_DEPRECATIONS_TO_EXCEPTIONS=false`

Comment (driesvints) [reply .github/workflows/tests.yml:162]: This is fine 👍

File: src/Illuminate/Console/Scheduling/ScheduleListCommand.php (hunk 2 above)
Comment (driesvints) [reply src/Illuminate/Console/Scheduling/ScheduleListCommand.php]: I tackled this differently by using an empty string fallback in the `mutexName` method of `CallbackEvent`.

**Pull request merged**